src/i2c.rs

```diff
@@ -297,67 +411,52 @@ macro_rules! hal {
 
                     let pclk = <$I2CX as RccBus>::Bus::get_frequency(&clocks).0;
 
-                    assert!(mode.get_frequency().0 <= 400_000);
-
                     let mut i2c = I2c { i2c, pins, mode, pclk };
-                    i2c.init();
+                    i2c.init(clocks);
                     i2c
                 }
 
                 /// Initializes I2C as master. Configures I2C_PRESC, I2C_SCLDEL,
                 /// I2C_SDAEL, I2C_SCLH, I2C_SCLL
                 ///
                 /// For now, only standard mode is implemented
-                fn init(&mut self) {
+                fn init(&mut self, _clocks: Clocks) {
```

Comment:
`_clocks` doesn't seem to be used. Why add it? To make sure that RCC config has been frozen?